packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx

```diff
@@ -0,0 +1,270 @@
+import React, { ReactNode, useEffect, useRef, useState } from 'react';
+import { Components, registerComponent } from '../../lib/vulcan-lib';
+import { AnalyticsContext, captureEvent } from '../../lib/analyticsEvents';
+import classNames from 'classnames';
+import { tagPostTerms } from '../tagging/TagPage';
+import { useMulti } from '../../lib/crud/withMulti';
+import debounce from 'lodash/debounce';
+
+
+const styles = (theme: ThemeType): JssStyles => ({
+  tabsSection: {
+    marginTop: 10,
+    marginBottom: 26,
+    [theme.breakpoints.down('sm')]: {
+      marginTop: 20,
+    },
+  },
+  tabsRow: {
+    position: 'relative',
+  },
+  tabsWindowContainer: {
+    position: 'relative',
+  },
+  tabsWindow: {
+    position: 'relative',
+    overflowX: 'scroll',
+    '&::-webkit-scrollbar': {
+      height: 0
+    }
+  },
+  tabsWindowFade: {
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    height: '100%',
+    width: 50,
+    content: "''",
+    background: `linear-gradient(to left, ${theme.palette.background.default}, transparent)`,
+    pointerEvents: 'none'
+  },
+  arrow: {
+    position: 'absolute',
+    top: 0,
+    height: 30,
+    width: 28,
+    color: theme.palette.grey[600],
+    paddingTop: 8,
+    paddingLeft: 6,
+    cursor: 'pointer',
+    '&:hover': {
+      color: theme.palette.grey[900],
+    },
+    '@media (max-width: 840px)': {
+      display: 'none',
+    },
+  },
+  arrowIcon: {
+    fontSize: 15
+  },
+  leftArrow: {
+    left: -30,
+  },
+  rightArrow: {
+    right: -30,
+  },
+  topicsBar: {
+    display: 'flex',
+    columnGap: 8,
+    whiteSpace: 'nowrap',
+    transition: 'transform 0.2s ease',
+    '@media (max-width: 840px)': {
+      columnGap: 6,
+    }
+  },
+  tab: {
+    backgroundColor: theme.palette.grey[300],
+    color: theme.palette.grey[900],
+    fontFamily: theme.typography.fontFamily,
+    fontSize: 14,
+    lineHeight: '23px',
+    fontWeight: '500',
+    padding: '4px 8px',
+    borderRadius: 6,
+    cursor: "pointer",
+    "&:hover": {
+      backgroundColor: theme.palette.grey[400],
+    },
+    '@media (max-width: 840px)': {
+      fontSize: 12,
+      lineHeight: '18px',
+      padding: '4px 6px',
+    }
+  },
+  activeTab: {
+    backgroundColor: theme.palette.grey[1000],
+    color: theme.palette.grey[0],
+    "&:hover": {
+      backgroundColor: theme.palette.grey[1000],
+    },
+  }
+})
+
+type TopicsBarTab = {
+  _id: string,
+  name: string,
+  shortName?: string|null
+}
+
+/**
+ * This handles displaying the main content on the EA Forum home page,
+ * which includes the topics bar and the topic-specific tabs.
+ * The "Frontpage" tab content comes from EAHome.
+ */
+const EAHomeMainContent = ({frontpageNode, classes}:{
+  frontpageNode: ReactNode,
```

Comment:
It'd probably be better to pass this in as an unrendered `ComponentType`, and then to actually render it here if it's needed. As it is at the moment, we're always rendering the "New & upvoted" markup even if we're displaying one of the tag tabs.